### PR TITLE
ENH Use appropriate formfields when auto-scaffolding files

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Issue
- https://github.com/silverstripe/.github/issues/262

### CODE REVIEW

File: src/File.php [hunk 1]
```diff
@@ -594,14 +595,44 @@ public function scaffoldFormFieldForHasOne(
         ?string $fieldTitle,
         string $relationName,
         DataObject $ownerRecord
-    ): FormField&FileHandleField {
+    ): FormField {
```
Comment (ChloeSartorelli) [inline src/File.php:598]: This is necessary for the change to `Folder`.

File: src/Folder.php [hunk 2]
```diff
@@ -94,6 +98,35 @@ public static function find_or_make($folderPath)
...
+
+    public function scaffoldFormFieldForManyMany(
+        string $relationName,
+        ?string $fieldTitle,
+        DataObject $ownerRecord,
+        bool &$includeInOwnTab
+    ): FormField {
+        $includeInOwnTab = false;
+        return TreeMultiselectField::create($relationName, $fieldTitle, static::class);
+    }
```
Comment (ChloeSartorelli) [inline src/Folder.php:128]: You don't upload folders. This makes a lot more sense as a way to select folders, rather than using a file selection UI.

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Tested locally, works well, except that if I have a relation that is an Image::class, the auto-scaffolded upload field allows me to select regular files like PDFs which aren't images. Applies to has_one, has_many and many_many relations.

**Comment by GuySartorelli** [comment]
When attempting to upload a PDF, it gives an error "Extension 'pdf' is not allowed"
image

This is the pre-existing behaviour for has_one auto-scaffolded upload fields.

**Comment by emteknetnz** [comment]
Weird, I must have mis-tested this, unable to replicate my issue

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**